Mainnet/AddressMapper/AddressMapper/AddressMapper.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using Stratis.SmartContracts;
+
+public class AddressMapper : SmartContract
+{
+    public Address GetSecondaryAddress(Address primary) => PersistentState.GetAddress($"SecondaryAddress:{primary}");
+    private void SetSecondaryAddress(Address primary, Address secondary) => PersistentState.SetAddress($"SecondaryAddress:{primary}", secondary);
+
+    private MappingInfo GetMappingInfo(Address secondary) => PersistentState.GetStruct<MappingInfo>($"MappingInfo:{secondary}");
+    private void SetMappingInfo(Address secondary, MappingInfo value) => PersistentState.SetStruct($"MappingInfo:{secondary}", value);
+
+    private void ClearMappingInfo(Address secondary) => PersistentState.Clear($"MappingInfo:{secondary}");
+
+    public Address Owner
+    {
+        get => PersistentState.GetAddress(nameof(Owner));
+        private set => PersistentState.SetAddress(nameof(Owner), value);
+    }
+
+    public AddressMapper(ISmartContractState smartContractState, Address owner) : base(smartContractState)
+    {
+        this.Owner = owner;
+    }
+
+    public void MapAddress(Address secondary)
+    {
```

Comment:
Should it be possible to map an address to itself? Ie. if `secondary == Message.Sender`?

Reply:
Well we haven't restricted to code for that case but it is not going to harm the contract. We add this assertion so will add extra gas for users who sent parameter correctly. 

Do you prefer to add assertion for this ?